internal/tasks/source_lister.go

```diff
@@ -53,9 +56,15 @@ func NewSourceLister(input *NewSrcListerInput) (SourceListerAPI, error) {
 	return srcLister, nil
 }
 
-func (s *sourceLister) Do() error {
-	if err := filepath.WalkDir(s.SrcRootDir, s.walkDirFunc); err != nil {
-		return err
+func (s *sourceLister) Do(withReferenceTime bool) error {
```

Comment:
```suggestion
func (s *sourceLister) Do(referenceTime *time.Time) error {
```

Reply:
We cannot avoid from putting `ReferenceTime` in `SourceListerInput`, because if it is only available via `Do()`'s input parameter, it will not be available to `walkDirFunc` and `walkDirWithFuncReferenceTime`, which can only have a `(path string, d fs.DirEntry, err error)` input signiture.